### PR TITLE
Adds support for Applications in any namespace

### PR DESCRIPTION
**Pull request by mwienk**
I've added support for namespaced applications in Argo CD (issue #1068 and #1038). I've added it all under a feature flag argocd.namespacedApps.

I do think it needs documentation for the functionality. At least the part enabling the functionality:
```yaml
[app-config.yaml]
argocd:
  namespacedApps: true
```

And how to use new annotation:
```yaml
[entity.yaml]
...
metadata:
  annotations:
    argocd/app-namespace: my-test-ns
```

should be available somwhere, but I'm not sure if you want it in the readme file or perhaps somewhere else to keep things clean.

#### :heavy_check_mark: Checklist

- [X] Added tests for new functionality and regression tests for bug fixes
- [X] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by unitygilles** [comment]
I have tested PR on our ArgoCD setup that uses apps in any namespaces and it works as expected. Good job @mwienk . @punkle / @kissmikijr  is this something you can review or provide insight into?

File: plugins/frontend/backstage-plugin-argo-cd/src/api/index.ts [hunk 1]
```diff
@@ -158,67 +181,86 @@ export class ArgoCDApiClient implements ArgoCDApi {
...
+      appNamespace: options.appNamespace,
+    });
     if (this.searchInstances) {
       return this.fetchDecode(
         `${proxyUrl}/argoInstance/${
           options.instance
-        }/applications/name/${encodeURIComponent(options.appName as string)}`,
+        }/applications/name/${encodeURIComponent(
+          options.appName as string,
+        )}?${query}`,
```
Comment (Zigu) [inline plugins/frontend/backstage-plugin-argo-cd/src/api/index.ts:205]: The ? before ${query} is wrong here, isn't it? the getQueryParams will add the ? automatically.

If yes, please remove.

Comment (mwienk) [reply plugins/frontend/backstage-plugin-argo-cd/src/api/index.ts]: Good catch, fixed it

**Comment by Xantier** [comment]
If this would have updates to README, it would likely be good to go

**Comment by kissmikijr** [comment]
@mwienk If you have time could you update the README for this change ?

**Comment by mwienk** [comment]
I've updated the README as requested.

**Review by Xantier — APPROVED** [review]

**Pull request merged**